flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -104,11 +105,38 @@ public static Schema convert(Schema baseSchema, TableSchema flinkSchema) {
     Types.StructType struct = convert(flinkSchema).asStruct();
     // reassign ids to match the base schema
     Schema schema = TypeUtil.reassignIds(new Schema(struct.fields()), baseSchema);
+    // reassign doc to match the base schema
+    schema = reassignDoc(schema, baseSchema);
```

Comment:
The reason I added this logic is:
https://github.com/apache/iceberg/pull/5376/files#diff-db9787684ca50d6d4fdfe8ee927613d2e8e5c29eaa894535d39147b4f631cb17R152-R157

If the doc of the field is different, it will not pass the validation.

Reply:
it seems that `reassignDoc` can be put inside `TypeUtil` like `reassignIds`